lib/veritech-server/src/server.rs

```diff
@@ -92,6 +94,26 @@ pub enum ServerError {
     WrongCycloneSpec(&'static str, Box<CycloneSpec>),
 }
 
+impl ServerError {
+    pub fn into_function_result<T>(self, execution_id: String) -> FunctionResult<T> {
+        FunctionResult::Failure(match self {
+            ServerError::Cancelled => FunctionResultFailure::new(
+                execution_id,
+                FunctionResultFailureError {
+                    kind: FunctionResultFailureErrorKind::CancelledExecution,
+                    message: self.to_string(),
+                },
+                timestamp(),
+            ),
+            _ => FunctionResultFailure::new_for_veritech_server_error(
+                execution_id,
+                self.to_string(),
+                timestamp(),
+            ),
+        })
+    }
+}
```

Comment:
As a heads up, `Cancelled` becomes `Killed` in #4438.

Reply:
Oh ok, I'll wait for that to merge before merging this one